tapio/parser/parser.py

```
@@ -0,0 +1,728 @@
+                # File is outside domain directory
+                self.logger.info(f"File outside domain dir, using base URL: {self.config.base_url}")
+                return self.config.base_url
+                
+            # Normalize path and construct URL
+            normalized_path = rel_path.replace("\\", "/")
+            constructed_url = urljoin(self.config.base_url, normalized_path)
+            self.logger.info(f"Constructed base URL: {constructed_url}")
+            return constructed_url
+            
+        except ValueError:
+            self.logger.warning(f"Error constructing URL from path, using base URL: {self.config.base_url}")
```

Ruff (0.11.9)

Line too long (108 > 100)

(E501)